wallet/difficulty.go

```diff
@@ -247,6 +250,287 @@ func (w *Wallet) nextRequiredPoWDifficulty(dbtx walletdb.ReadTx, header *wire.Bl
 	return nextDiffBits, nil
 }
 
+func (w *Wallet) loadCachedBlake3WorkDiffCandidateAnchor() *wire.BlockHeader {
+	w.cachedBlake3WorkDiffCandidateAnchorMu.Lock()
+	defer w.cachedBlake3WorkDiffCandidateAnchorMu.Unlock()
+
+	return w.cachedBlake3WorkDiffCandidateAnchor
+}
+
+func (w *Wallet) storeCachedBlake3WorkDiffCandidateAnchor(candidate *wire.BlockHeader) {
+	w.cachedBlake3WorkDiffCandidateAnchorMu.Lock()
+	defer w.cachedBlake3WorkDiffCandidateAnchorMu.Unlock()
+
+	w.cachedBlake3WorkDiffCandidateAnchor = candidate
+}
+
+// isBlake3PowAgendaForcedActive returns whether or not the agenda to change the
+// proof of work hash function to blake3, as defined in DCP0011, is forced
+// active by the chain parameters.
+func (w *Wallet) isBlake3PowAgendaForcedActive() bool {
+	const deploymentID = chaincfg.VoteIDBlake3Pow
+	version := voteVersion(w.chainParams)
```

Comment:
this probably introduces a bug when the vote version is bumped for the next series of deployments. it will no longer discover the deployment with the blake3 pow id.

Reply:
Yeah, it would.  This is handled in dcrd by using a map and ensuring that all vote IDs are globally unique.